tests/test_single_value_per_month.R

```diff
@@ -0,0 +1,483 @@
+  set.seed(123)
+
+  # Missing months
+  data <- runif(10, 0, 20)
+  dates <- seq(as.PCICt("2020-01-01", cal = "gregorian"), by = "2 months", length.out = 10)
+  result <- try(
+    scalar_obj <- climdexSingleMonthlyScalar.raw(data, dates, northern.hemisphere = TRUE, calendar = "gregorian"),
+    silent = TRUE
+  )
+
+  checkTrue(
+    !inherits(result, "try-error"),
+    "Function raised an error despite valid monthly data."
+  )
+  checkEquals(scalar_obj@dates[!is.na(scalar_obj@data)], dates, "Scalar object dates for non-NA data does not match input dates.")
+}
+
+climdex.pcic.test.SingleMonthlyScalar.raw.negative.values <- function() {
+  # Data with negative values
+  data <- runif(36, -50, 0)
+  dates <- seq(as.PCICt("2020-01-01", cal = "gregorian"), by = "month", length.out = 36)
+
+  scalar_obj <- climdexSingleMonthlyScalar.raw(data, dates, northern.hemisphere = TRUE, calendar = "gregorian")
+  scalar_max <- unname(compute.stat.scalar(scalar_obj, "max", "annual", FALSE))
+  scalar_min <- unname(compute.stat.scalar(scalar_obj, "min", "annual", FALSE))
+  scalar_sum <- unname(compute.stat.scalar(scalar_obj, "sum", "annual", FALSE))
+  checkEquals(max(scalar_max), max(data), "annual max stat for single monthly scalar was not equal to max of input data")
+  checkEquals(min(scalar_min), min(data), "annual min stat for single monthly scalar was not equal to min of input data")
+  checkEquals(sum(scalar_sum), sum(data), "annual sum stat for single monthly scalar was not equal to sum of input data")
+}
+
+climdex.pcic.test.SingleMonthlyVector.raw.cartesian <- function() {
+  set.seed(123)
+
+  # Data in cartesian format (x,y components)
+  x_comp <- runif(12, -10, 10)
+  y_comp <- runif(12, -10, 10)
+  dates <- seq(as.PCICt("2020-01-01", cal = "gregorian"), by = "month", length.out = 12)
+
+  vector_obj <- climdexSingleMonthlyVector.raw(
+    primary = x_comp,
+    secondary = y_comp,
+    dates = dates,
+    format = "cartesian",
+    northern.hemisphere = TRUE,
+    calendar = "gregorian"
+  )
+
+  # Verify components
+  checkEquals(length(vector_obj@primary), length(vector_obj@dates), "Cartesian vector primary length does not match dates.")
+  checkEquals(vector_obj@primary[!is.na(vector_obj@primary)], x_comp, "X component not stored correctly")
+  checkEquals(vector_obj@secondary[!is.na(vector_obj@secondary)], y_comp, "Y component not stored correctly")
+}
+
+climdex.pcic.test.SingleMonthlyScalar.large.dataset <- function() {
+  set.seed(123)
+
+  # Create 100 years of monthly data
+  n_months <- 100 * 12
+  data <- runif(n_months, 0, 20)
+  dates <- seq(as.PCICt("1920-01-01", cal = "gregorian"), by = "month", length.out = n_months)
+
+  scalar_obj <- climdexSingleMonthlyScalar.raw(
+    data = data,
+    dates = dates,
+    northern.hemisphere = TRUE,
+    calendar = "gregorian"
+  )
+
+  checkEquals(length(scalar_obj@data[!is.na(scalar_obj@data)]), n_months, "Large dataset not handled correctly")
+}
```

Comment:
Without myself checking for every possible error message, this looks thorough and like it covers all those error conditions. Well done.